### ISSUE TITLE
KeyError: 'content'

### ISSUE THREAD

**Issue by joreeves** [xjorma/EarthMeshHoudini#14]
Occasionally get this error for no reason. If I change some of the settings sometimes it goes away.

![image](https://github.com/xjorma/EarthMeshHoudini/assets/82180221/04df5e75-30d5-4a2f-99c5-fc7c7f3b76c4)

**Comment by Piminski** [comment]
Started getting similar issues as well. As you mentioned changing the parameters can resolve it but I cannot describe yet what exactly is the cause, seems to be min dist and/or max error... 

It may be something has changed on the Google side....

**Comment by joreeves** [comment]
I've found the problem. It's an edge case where there is only 1 child which has no "content" in it. I've created a pull request.

**Comment by Piminski** [comment]
![image](https://github.com/xjorma/EarthMeshHoudini/assets/84422903/f37e278e-89e4-4a0e-b8b8-cde13f565475)
I downloaded your new HDA and test hip file but I get this error

**Comment by joreeves** [comment]
@Piminski With the updates I made the "scripts" folder needs to be placed relative to your houdini file. Sorry I should've stated that (I guess that's a downside to moving the scripts outside of Houdini). Could you try opening the EarthMesh_Test.hip file and let me know if you get the same error there?

**Comment by xjorma** [comment]
I will take a look.

**Comment by xjorma** [comment]
I submitted a blind fix.

Sorry, @joreeves. I will investigate a bit more. Your fix is certainly more tested than mine. Your code is definitely cleaner, tests more edge cases elegantly, and it's a good idea to have Python as an external file (I dislike the fact that everything is embedded in a binary file in Houdini or Touchdesigner).

However, I wanted more time to see if external files are really a good idea for the end-user (ie; script folder relative path). Additionally, there are too many refactors. Your code is definitely better and easier to maintain, but I also have some local changes to make it work with Houdini Engine (within Unreal, for example).

Can someone validate the fix? or provide me a screen shot with the parameter that cause the issue? (don't forget to make sure your key is not visible)

**Comment by joreeves** [comment]
@xjorma all good. It may not be the best solution to have external files. I prefer them that way for tracking on github but obviously it means having to keep the files relative to the hda which is not ideal. Maybe there's a better solution out there that combines both.

**Comment by joreeves** [comment]
@xjorma I believe I found a good solution for this. I will create a new pull request. Feel free to use it if you like or re-implement if you'd prefer.

**Comment by xjorma** [comment]
@joreeves cool let me know

**Comment by Piminski** [comment]
@joreeves Thank you for your efforts. I am a bit new to github so I might be doing things wrong. I cloned the repo locally using command line but I seem to be getting the old version, I do not understand how to get your version for testing. I tried git clone on the pull request as well. Any advice is appreciated. I tried chatgpt but she was not helpful ;)

**Comment by joreeves** [comment]
@Piminski try this (assuming on windows):

First make sure there isn't a folder named EarthMeshHoudini in the location you want to clone to. Open a cmd prompt. You can run `cd PATH/TO/FOLDER` to change your directory you are installing to.

Then try running `git clone https://github.com/joreeves/EarthMeshHoudini.git`

If cloning isn't working you can also download a zip directly from github. Just click on the "code" button then select "Download ZIP"
![image](https://github.com/xjorma/EarthMeshHoudini/assets/82180221/70150ee6-9c8d-4b54-8282-721db156e6ef)

**Comment by Piminski** [comment]
@joreeves Thank you!

I think I have managed to get it working! Meshes are currently loading, hurrah!

One thing I was thinking is that the HDA starts loading meshes immediately so if the latitude changes it loads before one can change the longitude. I have been doing workarounds by disabling the node or removing the API key but I though an "load Mesh" button would be great. I even made a very basic/stupid HDA wrapping the Earthmesh in a subnet with a switch to a null. Maybe this a separet issue but as I mentioned I am not used to the git process yet.

Thanks to you and David again, this tool is so very useful.

**Comment by joreeves** [comment]
@Piminski Yeah, I noticed that as well and a button is a good idea. If you want to avoid it automatically loading for now a workaround I've been using is switching Houdini from "Auto Update" to "Manual" or your method sounds good as well.

**Comment by Piminski** [comment]
Hey @joreeves 

It was all working well but for some reason it has stopped working. I had a functioning setup using the Earthmesh2:0 hda where I cached everything out to bgeo.sc files, when I now try to regenerate it gives the same error I described before. I have the scripts folder in the project folder relative to the hip file.

I am sorry to bother you. 

Have you experienced anything similar? Has anything changed? Any advice is appreciated?


![Screenshot 2024-07-22 175208](https://github.com/user-attachments/assets/c2f96e12-0a18-49c2-a5ce-d8c8bd82dd90)
![Screenshot 2024-07-22 175840](https://github.com/user-attachments/assets/2999728f-3f7f-4fa3-9ea7-5020ae779c63)

**Comment by joreeves** [comment]
Hey, I haven’t changed anything. I’m guessing something in googles api
probably changed. I can take a look later and submit a fix.

**Comment by joreeves** [comment]
Hey @Piminski so I think it might be an installation issue. Sorry I changed things around a few times back when i was making these fixes. If you are using the latest version (which I believe you are) the scripts folder no longer needs to be relative to your project. It should live in a stationary folder. Here's the install instructions:

- Download the repo.
- Place the EarthMeshHoudini folder somewhere on your local drive. It's important that you do not install EarthMeshHoudini directly into your $HOME/houdiniXX.X directory or into any other default Houdini installation directory, or else it may not properly be loaded when you start Houdini.
- Configure the Google API Key in the EarthMesh.json file and change the EARTHMESH_PATH to the path where you placed the EarthMeshHoudini folder.
- Copy the EarthMesh.json file to your $HOME/houdiniXX.X/packages folder.

Could you check to see if you have the EarthMesh.json file in your `$HOME/houdiniXX.X/packages` folder? And if so, where does the **EARTHMESH_PATH** variable point to. That's where the scripts should live. Let me know if that helps.

**Comment by Piminski** [comment]
Thank you @joreeves!

That seemed to fix it. I had the hda inside the documents/Houdini. Now per your instruction it works again. I cannot understand why it worked before but anyways.

The API credential in the json it makes the autostarting-download even more of an issue but I just put a wrong code in there and let it fail when I put the node
